platform/flowglad-next/src/utils/bookkeeping/customerBilling.test.ts

```diff
@@ -582,6 +585,408 @@ describe('setDefaultPaymentMethodForCustomer', () => {
+          },
+          params.transaction
+        )
+      })
+
+      // Assert: pricingModel.products[].prices[] should contain only active prices
+      expect(billingState.pricingModel).toBeDefined()
+      expect(billingState.pricingModel.products).toHaveLength(2) // setupOrg + our test product
+      
+      // Find our test product
+      const testProduct = billingState.pricingModel.products.find(p => p.id === productWithMixedPrices.id)
+      expect(testProduct).toBeDefined()
+      expect(testProduct!.prices).toHaveLength(1) // Only active price should be returned
+      
+      const returnedPrice = testProduct!.prices[0]
+      expect(returnedPrice.id).toBe(activePrice.id)
+      expect(returnedPrice.active).toBe(true)
+      
+      // Verify inactive price is filtered out
+      const inactivePriceInResult = testProduct!.prices.find(p => p.id === inactivePrice.id)
+      expect(inactivePriceInResult).toBeUndefined()
+    })
+
+    it('should preserve subscription items with inactive prices (grandfathered state)', async () => {
+      // Action: Call customerBillingTransaction
+      const billingState = await adminTransaction(async (params) => {
+        return await customerBillingTransaction(
+          {
+            externalId: customer.externalId,
+            organizationId: organization.id,
+          },
+          params.transaction
+        )
+      })
+
+      // Assert: Subscription items with inactive prices should still be visible
+      expect(billingState.subscriptions).toBeDefined()
+      expect(billingState.subscriptions.length).toBeGreaterThanOrEqual(2)
+      
+      // Find subscription with active price
+      const subscriptionWithActivePrice = billingState.subscriptions.find(
+        sub => sub.priceId === activePrice.id
+      )
+      expect(subscriptionWithActivePrice).toBeDefined()
+      expect(subscriptionWithActivePrice?.priceId).toBe(activePrice.id)
+      
+      // Find subscription with inactive price (grandfathered state)
+      const subscriptionWithInactivePrice = billingState.subscriptions.find(
+        sub => sub.priceId === inactivePrice.id
+      )
+      expect(subscriptionWithInactivePrice).toBeDefined()
+      expect(subscriptionWithInactivePrice?.priceId).toBe(inactivePrice.id)
+      
+      // Both subscription items should be visible regardless of price active status
+      // This tests the grandfathered state where subscription items remain visible
+      // even if their associated price becomes inactive
+    })
+
+    it('should maintain all other billing data while filtering prices', async () => {
+      // Action: Call customerBillingTransaction
+      const billingState = await adminTransaction(async (params) => {
+        return await customerBillingTransaction(
+          {
+            externalId: customer.externalId,
+            organizationId: organization.id,
+          },
+          params.transaction
+        )
+      })
```

Comment:
always use `transaction` explicitly to enforce the readability-improving convention
```suggestion
      const billingState = await adminTransaction(async ({ transaction } ) => {
        return await customerBillingTransaction(
          {
            externalId: customer.externalId,
            organizationId: organization.id,
          },
          transaction
        )
      })
```